xun/functions/util.py

```diff
@@ -203,6 +203,51 @@ def assignment_target_names(t):
         raise TypeError('Unsupported target {}'.format(type(t)))
 
 
+def assignment_target_shape(t):
+    """Assignment target shape
+
+    Given a target expression node, of type ast.Tuple or ast.List,
+    return the shape of the target.
+
+    Examples
+    --------
+    >>> target = 'a, (b, c)'
+    >>> ast_target = ast.parse(target).body[0].value
+    >>> assignment_target_shape(ast_target)
+    (1, (2,))
+
+    """
+    assert isinstance(t, (ast.Tuple, ast.List))
+
+    inner_tuple = ()
+    count_Names = 0
```

Comment:
Do we need this? Or rather, what is the effect of this? Maybe we can make the intent more clear